Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabUndo.cpp

```diff
@@ -96,7 +99,62 @@ namespace AzToolsFramework
             m_instanceToTemplateInterface->PatchTemplate(m_redoPatch, m_templateId);
         }
 
-        // PrefabEntityUpdateUndo
+        // PrefabUndoRemoveEntities
+
+        PrefabUndoRemoveEntities::PrefabUndoRemoveEntities(const AZStd::string& undoOperationName)
+            : PrefabUndoBase(undoOperationName)
+        {
+        }
+
+        void PrefabUndoRemoveEntities::Capture(
+            const AZStd::vector<PrefabDom>& entityDoms, const AZStd::vector<AZStd::string>& patchPaths, TemplateId templateId)
```

Comment:
Since there is a 1-1 correspondance with entityDoms and patchPaths, you can use an AZStd::pair here and use the range based for loop

Reply:
Good suggest!